iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/execution/config/executor/ClusterConfigTaskExecutor.java

```diff
@@ -2074,6 +2074,16 @@ public SettableFuture<ConfigTaskResult> alterPipe(final AlterPipeStatement alter
     final Map<String, String> connectorAttributes;
     try {
       if (!alterPipeStatement.getExtractorAttributes().isEmpty()) {
+        // We simply don't allow to alter external sources
```

Comment:
Now the implementation is "cannot alter source if the original source is external", but it seems that the intended one is "cannot alter source type"? Now you can alter from iotdb-source to another, and cannot alter the "mqtt-source"'s own variables... Not perfect